File: Labs/week4/C_queue_simulation.py
from collections import deque

n = int(input())
q = deque()
for _ in range(n):
    parts = input().strip().split()
    op = parts[0]
    if op == "push":
        q.append(int(parts[1]))
    elif op == "query":
        print(q[0])
    elif op == "empty":
        print("YES" if not q else "NO")
    elif op == "pop":
        q.popleft()

File: Labs/week4/C_queue_simulation.cpp
#include <iostream>
#include <queue>
#include <string>
using namespace std;

int main() {
    int n;
    cin >> n;
    queue<int> q;
    string op;
    int x;
    while (n--) {
        cin >> op;
        if (op == "push") {
            cin >> x;
            q.push(x);
        } else if (op == "query") {
            cout << q.front() << endl;
        } else if (op == "empty") {
            cout << (q.empty() ? "YES" : "NO") << endl;
        } else if (op == "pop") {
            q.pop();
        }
    }
    return 0;
}
